src/seccomp-support.c

```diff
@@ -500,6 +501,71 @@ static void preprocess_filter(FILE * f, struct preprocess *p)
 	return;
 }
 
+uint32_t get_hostarch(void)
+{
+	struct utsname uts;
+	if (uname(&uts) < 0)
+		die("uname() failed");
+
+	if (strcmp(uts.machine, "i686") == 0)
```

Comment:
This block can be a small dedicated function that could be unit tested for sanity. I can gladly patch this.